blocks/components/editable/index.js

```diff
@@ -188,18 +207,26 @@ export default class Editable extends wp.element.Component {
 	}
 
 	onNodeChange( { parents } ) {
-		const formats = parents.reduce( ( result, node ) => {
+		let alignment = null;
+		const formats = {};
+
+		parents.forEach( ( node ) => {
 			const tag = node.nodeName.toLowerCase();
 
 			if ( formatMap.hasOwnProperty( tag ) ) {
-				result[ formatMap[ tag ] ] = true;
+				formats[ formatMap[ tag ] ] = true;
 			}
 
-			return result;
-		}, {} );
+			if ( tag === 'p' ) {
```

Comment:
This is not a restriction, that's where TinyMCE adds the alignment style. We can check in all the parent blocks, and then we'll have to decide for a priority (the closest maybe).